packages/pillarbox-debug-panel/src/pillarbox-debug-panel.js

```diff
@@ -0,0 +1,221 @@
+import videojs from 'video.js';
+import { version } from '../package.json';
+import './metric-component.js';
+
+/**
+ * @ignore
+ * @type {typeof import('video.js/dist/types/component').default}
+ */
+const Component = videojs.getComponent('Component');
+
+/**
+ * Represents a PillarboxDebugPanel component for the Video.js player.
+ *
+ * The PillarboxDebugPanel is a high-level component that serves as a container
+ * for multiple MetricComponent instances. Each MetricComponent is responsible
+ * for displaying a specific video playback metric, optionally with a graphical
+ * representation (using a GraphComponent).
+ */
+class PillarboxDebugPanel extends Component {
+  update_  = () => this.update();
+
+  /**
+   * Creates an instance of a PillarboxDebugPanel.
+   *
+   * @param {import('video.js/dist/types/player.js').default} player - The Video.js player instance.
+   * @param {Object} options - Configuration options for the PillarboxDebugPanel.
+   */
+  constructor(player, options) {
+    super(player, options);
+
+    this.player().one('loadeddata', this.update_);
+    this.player().on('timeupdate', this.update_);
+  }
+
+  dispose() {
+    this.player().off('loadeddata', this.update_);
+    this.player().off('timeupdate', this.update_);
+  }
+
+  /**
+   * Updates all MetricComponent instances within this panel.
+   */
+  update() {
+    this.children().forEach(child => {
+      if (child.update) {
+        child.update();
+      }
```

Comment:
Below you short-circuited, but here you found that the update would be nice and warm wrapped up in this `if`?